app/scripts/modules/amazon/src/securityGroup/configure/configSecurityGroup.mixin.controller.js

```diff
@@ -326,6 +323,7 @@ module(AMAZON_SECURITYGROUP_CONFIGURE_CONFIGSECURITYGROUP_MIXIN_CONTROLLER, [
     };
 
     ctrl.addRule = function(ruleset) {
+      ruleset = ruleset || [];
```

Comment:
I know this is a little weird, but we are mutating the original array so `(ruleset || [])` doesn't do the job.

Reply:
Weird - isn't this just changing the reference to the array within the function? I don't understand how this would set the ruleset on the security group.